packages/shared/src/hooks/useTutorial.spec.tsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import { renderHook, RenderResult } from '@testing-library/react-hooks';
+import { QueryClient, QueryClientProvider } from 'react-query';
+import { waitFor } from '@testing-library/react';
+import { TutorialKey, UseTutorial, useTutorial } from './useTutorial';
+
+describe('useTutorial hook', () => {
+  const createWrapper = ({ client }: { client: QueryClient }) => {
+    const Wrapper = ({ children }) => {
+      return (
+        <QueryClientProvider client={client}>{children}</QueryClientProvider>
+      );
+    };
+
+    return Wrapper;
+  };
+
+  const waitForCompletedValue = ({
+    client,
+    result,
+    toBe,
+  }: {
+    client: QueryClient;
+    result: RenderResult<UseTutorial>;
+    toBe: boolean;
+  }) => {
+    return waitFor(() => {
+      const value = client.getQueryData([
+        `tutorial-${TutorialKey.SEEN_NEW_SQUAD_TOOLTIP_KEY}`,
+        false,
+      ]);
+
+      expect(value).toBe(toBe);
+      expect(result.current.isCompleted).toBe(toBe);
+    });
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should return the tutorial state', async () => {
+    const client = new QueryClient();
+    const { result } = renderHook(
+      () => useTutorial({ key: TutorialKey.SEEN_NEW_SQUAD_TOOLTIP_KEY }),
+      { wrapper: createWrapper({ client }) },
+    );
+
+    expect(result.current.isActive).toBe(false);
+    await waitForCompletedValue({ client, result, toBe: false });
```

Comment:
Had to use this because `usePersistentContext` and I think the `idb-keyval` lib under the hood had some weird promise handling and they would leak after the tests have completed. `await new Promise(process.nextTick)` did not work either.